lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/YarnSubmissionHelper.java

```diff
@@ -279,6 +282,13 @@ public void submit() throws IOException, YarnException {
     }
 
     this.yarnClient.submitApplication(applicationSubmissionContext);
+
+    if (this.isUnmanaged) {
+      // For Unmanaged AM mode, add a new app token to the
+      // current process so it can talk to the RM as an AM.
+      final Token<AMRMTokenIdentifier> token = this.yarnClient.getAMRMToken(this.applicationId);
+      this.tokenProvider.addTokens(UserCredentialSecurityTokenProvider.serializeToken(token));
```

Comment:
@jwang98052 the token we get here is the *Driver's* token. In the regular ("managed AM") mode, we don't need that token on the client side, since the driver runs somewhere else on the cloud. For the Unmananged AM mode, though, we need to obtain the Driver's token explicitly and add it to the current process credentials so we can run the Driver in the same process where the Client is.